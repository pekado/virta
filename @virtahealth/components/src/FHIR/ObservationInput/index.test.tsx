import * as React from "react";
import { createComponentWithVirtaContext } from "../../test";
import { FhirObservationInput } from "./index";

it("renders the FHIR observation input", () => {
  const FhirObservationInputContainer = createComponentWithVirtaContext(
    <FhirObservationInput
      observation={{
        code: {
          text: "Glucose",
        },
        status: "final",
        valueQuantity: {
          unit: "mg/ml",
        },
      }}
      setObservation={jest.fn}
    />
  );
  expect(FhirObservationInputContainer).toMatchSnapshot();
});
