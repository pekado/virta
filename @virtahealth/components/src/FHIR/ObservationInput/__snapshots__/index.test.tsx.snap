// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FHIR observation input 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginRight": 0,
          "marginTop": 0,
          "minWidth": 205,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(36, 38, 43)",
            "fontFamily": "Whitney-Book",
            "fontSize": 12,
            "fontWeight": "500",
            "letterSpacing": 0,
            "lineHeight": 12,
          },
        ]
      }
    >
      Glucose
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "display": "flex",
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        editable={true}
        isFocused={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(255,255,255)",
              "borderColor": "rgb(214, 217, 221)",
              "borderRadius": 6,
              "borderStyle": "solid",
              "borderWidth": 2,
              "display": "flex",
              "flexBasis": "auto",
              "flexGrow": 1,
              "flexShrink": 0,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          isFocused={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(0,0,0)",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Whitney-Book",
                "fontSize": 20,
                "minHeight": 40,
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
              },
            ]
          }
          value=""
        />
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(36, 38, 43)",
                "fontFamily": "Whitney-Book",
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 20,
                "textDecorationColor": "black",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
              },
              Object {
                "alignItems": "center",
                "color": "rgb(89, 94, 101)",
                "display": "flex",
                "flexBasis": "auto",
                "fontFamily": "Whitney-Book",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0,
                "lineHeight": 18,
                "minHeight": 40,
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
              },
            ]
          }
        >
          mg/ml
        </Text>
      </View>
    </View>
  </View>
</View>
`;
